From 601336c9a5d6f35cc7364d415bf027ff1e70cf69 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 24 Nov 2016 10:10:59 +0200
Subject: [PATCH 1124/2241] fix: net: mvpp2x: disable mtu change for netmap
 port

- When a port is configured for netmap, disable option to
  change MTU size since netmap uses static buffers

Change-Id: Ic478db1737d02f0bb03b1488efb834cdac2c16d3
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34040
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 695e45e..7cda3b13 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -3771,6 +3771,13 @@ static int mv_pp2x_change_mtu(struct net_device *dev, int mtu)
 	struct mv_pp2x_port *port = netdev_priv(dev);
 	int err;
 
+#ifdef DEV_NETMAP
+	if (port->flags & MVPP2_F_IFCAP_NETMAP) {
+		netdev_err(dev, "MTU can not be modified for port configured to Netmap mode\n");
+		return -EPERM;
+	}
+#endif
+
 	mtu = mv_pp2x_check_mtu_valid(dev, mtu);
 	if (mtu < 0) {
 		err = mtu;
-- 
2.7.4

